ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,185 @@
+
+import type ConfigModel from 'vault/models/azure/config';
+import type IdentityOidcConfigModel from 'vault/models/identity/oidc/config';
+import type Router from '@ember/routing/router';
+import type StoreService from 'vault/services/store';
+import type VersionService from 'vault/services/version';
+import type FlashMessageService from 'vault/services/flash-messages';
+
+/**
+ * @module SecretEngineConfigureAzure component is used to configure the Azure secret engine
+ * For enterprise users, they will see an additional option to config WIF attributes in place of Azure account attributes.
+ * If the user is configuring WIF attributes they will also have the option to update the global issuer config, which is a separate endpoint named identity/oidc/config.
+ * @example
+ * <SecretEngine::ConfigureAzure
+    @model={{this.model.azure-config}}
+    @backendPath={{this.model.id}}
+    @issuerConfig={{this.model.identity-oidc-config}}
+    />
+ * 
+ * @param {object} model - Azure config model
+ * @param {string} backendPath - name of the Azure secret engine, ex: 'azure-123'
+ * @param {object} issuerConfigModel - the identity/oidc/config model
+ */
+
+interface Args {
+  model: ConfigModel;
+  issuerConfig: IdentityOidcConfigModel;
+  backendPath: string;
+}
+
+export default class ConfigureAzureComponent extends Component<Args> {
+  @service declare readonly router: Router;
+  @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked accessType = 'azure';
+  @tracked errorMessage = '';
+  @tracked invalidFormAlert = '';
+  @tracked saveIssuerWarning = '';
+
+  disableAccessType = false;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    // the following checks are only relevant to existing enterprise configurations
+    if (this.version.isCommunity && this.args.model.isNew) return;
+    const { isWifPluginConfigured, isAzureAccountConfigured } = this.args.model;
+    this.accessType = isWifPluginConfigured ? 'wif' : 'azure';
+    // if there are either WIF or azure attributes, disable user's ability to change accessType
+    this.disableAccessType = isWifPluginConfigured || isAzureAccountConfigured;
+  }
+
+  get modelAttrChanged() {
+    // "backend" dirties model state so explicity ignore it here
+    return Object.keys(this.args.model?.changedAttributes()).some((item) => item !== 'backend');
+  }
+
+  get issuerAttrChanged() {
+    return this.args.issuerConfig?.hasDirtyAttributes;
+  }
+
+  @action continueSubmitForm() {
+    this.saveIssuerWarning = '';
+    this.save.perform();
+  }
+
+  // check if the issuer has been changed to show issuer modal
+  // continue saving the configuration
+  submitForm = task(
+    waitFor(async (event: Event) => {
+      event?.preventDefault();
+      this.resetErrors();
+      const { issuerConfig } = this.args;
+
+      if (issuerConfig?.hasDirtyAttributes) {
```

Comment:
Could you reuse the getter above here?
```suggestion
      if (this.issuerAttrChanged) {
```

Reply:
Good catch. Thanks!